ignore.go

```diff
@@ -0,0 +1,44 @@
+package git
+
+/*
+#include <git2.h>
+*/
+import "C"
+import (
+	"runtime"
+)
+
+func (v *Repository) AddIgnoreRule(rules string) error {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ret := C.git_ignore_add_rule(v.ptr, C.CString(rules))
```

Comment:
This (and the one below) leaks the string. You need to keep it in a variable and free it after it's done via `C.free()`, which is usually done in a `defer`. Check out the other places in the library where we do this for examples.